clusterutil/clusterutil.go

```diff
@@ -0,0 +1,58 @@
+package clusterutil
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	// MetadataClusterVerificationLabelKey is the key of the cluster verification label gRPC metadata.
+	MetadataClusterVerificationLabelKey = "x-cluster"
+
+	ReasonClient = "client_check_failed"
+	ReasonServer = "server_check_failed"
+)
+
+var (
+	ErrNoClusterVerificationLabel         = fmt.Errorf("no cluster verification label in context")
+	errDifferentClusterVerificationLabels = func(clusterIDs []string) error {
+		return fmt.Errorf("gRPC metadata should contain exactly 1 value for key %q, but it contains %v", MetadataClusterVerificationLabelKey, clusterIDs)
+	}
+)
+
+func NewIncomingContext(containsRequestCluster bool, requestCluster string) context.Context {
```

Comment:
I don't see where is this used, and why we need a separate flag, can we just check `requestCluster != ""`?

Reply:
Discussed offline: it's used in tests. Then lets move it to the tests. If it's used in multiple packages, then just copy it, as [a little copying is better than a little dependency](https://go-proverbs.github.io/).

If we keep it here, next person who comes will not know whether it's used by some dependency of `dskit`.